### PR TITLE
WD-8065 - Update facades for Juju 3.3

### PR DESCRIPTION
**Pull request by huwshimi**
# Done

- Updated the schema.json for Juju 3.3.
- Built the new facades.
- Updated the package version.

# Details

https://warthogs.atlassian.net/browse/WD-8065

### CODE REVIEW

File: api/client.ts [hunk 1]
```diff
@@ -271,7 +271,7 @@ class Client {
       "user-data": "",
     };
     const url = this._transport._ws.url;
-    const origin = new URL(url).origin;
+    const origin = url;
```
Comment (huwshimi) [inline api/client.ts:274]: This was a drive-by fix to improve macaroon handling with the new JIMM.

File: api/facades/application/ApplicationV19.ts
Comment (huwshimi) [inline api/facades/application/ApplicationV19.ts:1]: Everything inside the api folder is built automatically by the script.

File: README.md
Comment (huwshimi) [inline README.md:1]: These changes are made automatically by the script.

File: generator/schema/schema.json
Comment (huwshimi) [inline generator/schema/schema.json:1]: This file is copied from Juju.

File: package.json [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+  "version": "4.1.0",
   "description": "Juju API client",
   "main": "dist/api/client.js",
   "types": "dist/api/client.d.ts",
   "type": "module",
   "scripts": {
     "build": "tsc",
     "clean": "rm -rf node_modules dist",
-    "generate-facades": "SCHEMA=generator/schema/schema.json JUJU_VERSION=$(cat generator/schema/juju-version.txt) JUJU_GIT_SHA=$(cat generator/schema/juju-git-sha.txt) ts-node generator/index.ts",
+    "generate-facades": "SCHEMA=generator/schema/schema.json JUJU_VERSION=$(cat generator/schema/juju-version.txt) JUJU_GIT_SHA=$(cat generator/schema/juju-git-sha.txt) node --no-warnings=ExperimentalWarning --loader ts-node/esm generator/index.ts",
```
Comment (huwshimi) [inline package.json:11]: This change was required as ts-node was having issues with Node > v16.

**Review by vladimir-cucu — APPROVED** [review]
Everything works as expected when testing with Juju Dashboard :rocket:
